scripts/src/java/org/oppia/android/scripts/maven/GenerateLicenseTexts.kt

```diff
@@ -0,0 +1,435 @@
+import javax.xml.transform.dom.DOMSource
+import javax.xml.transform.stream.StreamResult
+
+private const val MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE =
+  "maven_dependencies.textproto is not up-to-date"
+
+/**
+ * Script to extract the licenses for the third-party Maven dependencies (direct and indirect both)
+ * on which Oppia Android depends.
+ *
+ * Usage:
+ *   bazel run //scripts:generate_license_texts -- <path_to_directory_values>
+ *     <path_to_maven_dependenices.pb>
+ *
+ * Arguments:
+ * - path_to_directory_values: directory path to the values folder of the Oppia Android repository.
+ * - path_to_maven_dependenices.pb: relative path to the maven_dependencies.pb
+
+ * Example:
+ *   bazel run //scripts:generate_license_texts -- $(pwd)/app/src/main/res/values
+ *   scripts/assets/maven_dependencies.pb
+ */
+fun main(args: Array<String>) {
+  GenerateLicenseTexts(LicenseFetcherImpl()).main(args)
+}
+
+/**
+ * Wrapper class to pass dependencies to be utilized by the the main method to generate license
+ * texts of the dependencies.
+ */
+class GenerateLicenseTexts(
+  private val licenseFetcher: LicenseFetcher
+) {
+
+  /** Generates a resource xml file that contains license texts of the third-party dependencies. */
+  fun main(args: Array<String>) {
+    if (args.size < 2) {
+      println(
+        """
+        Usage: bazel run //scripts:generate_license_texts -- <path_to_directory_values>
+        <path_to_pb_file>  
+        """.trimIndent()
+      )
+      throw Exception("Too few arguments passed.")
+    }
+    val pathToValuesDirectory = args[0]
+    val pathToThirdPartyDependenciesXml = "$pathToValuesDirectory/third_party_dependencies.xml"
+    val pathToMavenDependenciesPb = args[1]
+
+    val mavenDependencyList = retrieveMavenDependencyList(pathToMavenDependenciesPb)
+    if (mavenDependencyList.isEmpty()) {
+      throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+    }
+    val copyrightLicenseSet = retrieveAllLicensesSet(mavenDependencyList)
+    val dependencyList =
+      retrieveDependencyList(mavenDependencyList)
+
+    val dependencyNamesList = retrieveArtifactsNamesList(dependencyList)
+    val dependencyVersionsList = retrieveArtifactsVersionsList(dependencyList)
+
+    writeThirdPartyDependneciesXml(
+      pathToThirdPartyDependenciesXml,
+      dependencyNamesList,
+      dependencyVersionsList,
+      copyrightLicenseSet,
+      dependencyList
+    )
+
+    println("\nScript execution completed successfully.")
+  }
+
+  /**
+   * Retrieve the list of maven dependencies from maven_dependencies.textproto.
+   *
+   * @param pathToPbFile path to the pb file to be parsed
+   *
+   * @return list of [MavenDependency]
+   */
+  private fun retrieveMavenDependencyList(pathToPbFile: String): List<MavenDependency> {
+    return parseTextProto(
+      pathToPbFile,
+      MavenDependencyList.getDefaultInstance()
+    ).mavenDependencyList
+  }
+
+  /**
+   * Helper function to parse the textproto file to a proto class.
```

Comment:
This & the helper name don't match the implementation--it's parsing a binary proto, not a textproto.

Reply:
Fixed.